doc/md/knowledge/postgres/serial-columns.md

```diff
@@ -0,0 +1,258 @@
+```
+
+As you can see, each serial column was created as non-nullable integer with a default value set to next sequence
+value.
+
+:::info
+Note that `nextval` increments the sequence by 1 and returns its value. Thus, the first call to
+`nextval('serials_c1_seq')` returns 1, the second returns 2, etc.
+:::
+
+### `ALTER COLUMN` type to serial
+
+Sometimes, it is necessary to change the column type from `integer` type to `serial`. However, as mentioned above, the
+`serial` type is not a true type, and therefore, the following commands will fail:
+
+```sql
+CREATE TABLE t(
+    c integer not null primary key
+);
+
+ALTER TABLE t ALTER COLUMN c TYPE serial;
+// highlight-next-line-error-message
+ERROR: type "serial" does not exist
+```
+
+We can achieve this by manually creating a [sequence](https://www.postgresql.org/docs/current/sql-createsequence.html)
+owned by the column `c`, and setting the column `DEFAULT` value to the incremental counter of the sequence using the
+[`nextval`](https://www.postgresql.org/docs/current/functions-sequence.html) function.
+
+:::note
+Note that, it is recommended to follow the PostgreSQL naming format (i.e. `<table>_<column>_seq`)
+when creating the sequence as some database tooling know to detect such columns as "serial columns".
+:::
+
+```sql
+-- Create the sequence.
+CREATE SEQUENCE "public"."t_c_seq" OWNED BY "public"."t"."c";
+
+-- Assign it to the table default value.
+ALTER TABLE "public"."t" ALTER COLUMN "c" SET DEFAULT nextval('"public"."t_c_seq"');
+```
+
+### Update the sequence value
+
+When a sequence is created, its value starts from 0 and the first call to `nextval` returns 1. Thus, in case the column
+`c` from the example above already contains values, we may face a constraint error on insert when the sequence number
+will reach to the minimum value of `c`. Let's see an example:
+
+```sql
+SELECT "c" FROM "t";
+// highlight-start
+ c 
+---
+ 2
+ 3
+// highlight-end
+ 
+-- Works!
+INSERT INTO "t" DEFAULT VALUES;
+-- Fails!
+INSERT INTO "t" DEFAULT VALUES;
+// highlight-next-line-error-message
+ERROR:  duplicate key value violates unique constraint "t_pkey"
+// highlight-next-line-error-message
+DETAIL:  Key (c)=(2) already exists.
+```
+
+We can work around this by setting the sequence current value to the maximum value of `c`, so the following call to
+`nextval` will return `MAX(c)+1`, the one after `MAX(c)+2`, etc.
+
+```sql
+SELECT setval('"public"."t_c_seq"', (SELECT MAX("c") FROM "t"));
+// highlight-start
+ setval 
+--------
+   3
+// highlight-end
+
+-- Works!
+INSERT INTO "t" DEFAULT VALUES;
+SELECT "c" FROM "t";
+// highlight-start
+ c 
+---
+ 2
+ 3
+ 4
+// highlight-end
+```
+
+### Managing Serial Columns with Atlas
+
+Atlas makes it easier to define and manipulate columns of `serial` types. Let's use the
+[`atlas schema inspect`](../../cli/reference.md#atlas-schema-inspect) command to get a representation in Atlas 
+the Atlas language of the table we created above:
```

Comment:
'the Atlas language' sounds kind of weird, I'd write in the Atlas HCL format or something